ingest/src/main/java/com/microsoft/azure/kusto/ingest/QueuedIngestClient.java

```diff
@@ -59,6 +59,15 @@
         this.azureStorageClient = azureStorageClient;
     }
 
+    public static String generateDmUriSuggestion(URIBuilder existingEndpoint) {
+        if (existingEndpoint.getHost().startsWith(INGEST_PREFIX))
+        {
+            throw new IllegalArgumentException("The URI is already a DM URI");
```

Comment:
1) This is specifically a URL, not just a URI
2) It'd be more accurate to say, "The URL is already formatted as the suggested DM endpoint, so no suggestion can be made"